x/merkledb/db.go

```diff
@@ -978,7 +978,7 @@ func (db *merkleDB) VerifyChangeProof(
 	switch {
 	case proof.Empty():
 		return ErrNoMerkleProof
-	case len(end) > 0 && len(proof.EndProof) == 0:
+	case len(end) > 0 && len(proof.KeyChanges) == 0 && len(proof.EndProof) == 0:
```

Comment:
should this be (len(end) > 0 || len(proof.KeyChanges) == 0) && len(proof.EndProof) == 0